subsys/llext/llext_link.c

```diff
@@ -177,39 +180,51 @@ int llext_link(struct llext_loader *ldr, struct llext *ext, bool do_local)
 			continue;
 		}
 
+		if (shdr->sh_info >= ldr->sect_cnt ||
+		    shdr->sh_size % shdr->sh_entsize != 0) {
```

Comment:
I don't understand this check... AFAIU, this checks if the section size is aligned to the ELF Shdr size?
Can you point out where in the ELF specification this is required?

Reply:
In fact it checks that the relocation section size is a multiple of _its entry's size_ (in this case, `elf_rel_t` or `elf_rela_t` structures). 
I am not sure this is _enforced_ by the spec, but why would it be different? It would be weird to have a section holding 5 and a half relocations :slightly_smiling_face: as you can have "empty space" in the ELF, not even alignment is a consideration. It's only an easy extra sanity filter on those values.

Reply:
Disregard my comment, I have mistakenly interpreted the `shdr->sh_entsize` as the `e_shentsize` field from ELF header, hence my confusion :sweat: 

The check indeed makes complete sense, it would definitely be an error if the section size was not aligned to the size of elements it's supposed to contain.
However, in this case, I think `shdr->sh_entsize` should be checked to be equal to `sizeof(elf_rel_t)`/`sizeof(elf_rela_t)` (or at least `!= 0`) *before the modulo*, otherwise a division by 0 could occur.

Reply:
At some point, we should introduce a test that attempts to load hand-crafted ELFs with such anomalies to ensure the loader's robustness.
Of course, this is outside this PR's scope :smile: